testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/ha/ReplicatedFailoverTestCase.java

```diff
@@ -115,4 +110,46 @@ private void configureCluster(ModelControllerClient client) throws Exception {
         operation.get(VALUE).set("clusteruser");
         execute(client, operation);
     }
+
+    @Override
+    protected void testMasterInSyncWithReplica(ModelControllerClient client) throws Exception {
+        ModelNode operation = Operations.createReadAttributeOperation(
+                PathAddress.parseCLIStyleAddress("/subsystem=messaging-activemq/server=default/ha-policy=replication-master").toModelNode(),
+                "synchronized-with-backup");
+        boolean synced = false;
+        long start = System.currentTimeMillis();
+        while (!synced && (System.currentTimeMillis() - start < 10000)) {
+            synced = execute(client, operation).asBoolean();
+        }
+        assertTrue(synced);
+    }
+
+    @Override
+    protected void testSlaveInSyncWithReplica(ModelControllerClient client) throws Exception {
+        ModelNode operation = Operations.createReadAttributeOperation(
+                PathAddress.parseCLIStyleAddress("/subsystem=messaging-activemq/server=default/ha-policy=replication-slave").toModelNode(),
+                "synchronized-with-live");
+        assertTrue(execute(client, operation).asBoolean());
+    }
+
+    @Override
+    protected void testMasterOutofSyncWithReplica(ModelControllerClient client) throws Exception {
+        ModelNode operation = Operations.createReadAttributeOperation(
+                PathAddress.parseCLIStyleAddress("/subsystem=messaging-activemq/server=default/ha-policy=replication-master").toModelNode(),
+                "synchronized-with-backup");
+        boolean synced = false;
```

Comment:
We should start with synced = true since we check for false here.

Reply:
nope as then it would be false on the first check thus getting out of the loop

Reply:
well, `!synced` could be changed to`synced` in the loop

Reply:
nope because on 2nd call it would be false and going out